testing/trino-tests/src/test/java/io/trino/tests/BaseQueryAssertionsTest.java

```diff
@@ -344,4 +344,47 @@ public void testIsNotFullyPushedDown()
                                 "\n" +
                                 "Output[columnNames = [name]]\n");
     }
+
+    @Test
+    public void testProjectedColumns()
```

Comment:
The test case fails when running against 'TestLocalQueryAssertions'.

```
java.lang.IllegalStateException: Column names are unknown

	at com.google.common.base.Preconditions.checkState(Preconditions.java:502)
	at io.trino.testing.MaterializedResult.getColumnNames(MaterializedResult.java:152)
	at io.trino.sql.query.QueryAssertions$QueryAssert.validateIfColumnsPresent(QueryAssertions.java:372)
	at io.trino.sql.query.QueryAssertions$QueryAssert.projected(QueryAssertions.java:338)
	at io.trino.tests.BaseQueryAssertionsTest.testProjectedColumns(BaseQueryAssertionsTest.java:352)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
```

Reply:
let me know if you need help with local query runner

Reply:
Captured column names for `LocalQueryRunner` in https://github.com/trinodb/trino/pull/16609/commits/457ce694702dd13209aae7aa2e2ccb17065d635e. Now test is passing.